### PR TITLE
Bug | Error message being swallowed in azure_rm_common.py

### LINKED ISSUE #31561: Bug | Error message being swallowed in azure_rm_common.py
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

azure_rm_common.py#L385

##### ANSIBLE VERSION

```
2.4
```

##### CONFIGURATION

None

##### OS / ENVIRONMENT

Ubuntu 16.04 LTS

##### SUMMARY

azure_rm_common is swallowing error message. What I see after executing azure module is
```
The full traceback is:
  File "/tmp/ansible_nL3Eo3/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py", line 365, in get_resource_group
    return self.rm_client.resource_groups.get(resource_group)
  File "/usr/local/lib/python2.7/dist-packages/azure/mgmt/resource/resources/v2017_05_10/operations/resource_groups_operations.py", line 306, in get
    f.write(response.raw)

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            ...
        }
    },
    "msg": "Error retrieving resource group ResourceGroup1 - expected a string or other character buffer object"
}
```
So it sounds that there is a problem with `ResourceGroup1`. But digging deeper, HTTP Resposne looks like:
```
403 Forbidden
{"error":{"code":"AuthorizationFailed","message":"The client 'xxx' with object id 'xxx' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/read' over scope '/subscriptions/yyy/resourcegroups/ResourceGroup1'."}}
```
So I understand that we might not want to log client/object or subscription IDs, but it would be useful to at least see 403 Forbidden, possibly together with some parts of the body.

`CloudError` in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L384 contains the whole HTTP Response object, so we could extract more output to show for debugging purposes.

### PR DESCRIPTION
**Pull request by ZhijunZhao**
##### SUMMARY

Fix #31561 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

azure_rm_common

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = None
  configured module search path = [u'/Users/zhijzhao/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

##### ADDITIONAL INFORMATION

```

```

@yuwzho @zikalino @yaweiw

### CODE REVIEW

**Comment by yuwzho** [comment]
shipit

File: lib/ansible/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -415,8 +415,8 @@ def get_resource_group(self, resource_group):
         '''
         try:
             return self.rm_client.resource_groups.get(resource_group)
-        except CloudError:
-            self.fail("Parameter error: resource group {0} not found".format(resource_group))
+        except CloudError as cloud_error:
+            self.fail("Error retrieving resource group {0} - {1}".format(resource_group, cloud_error.message))
```
Comment (jborean93) [inline lib/ansible/module_utils/azure_rm_common.py:419]: Is there a reason why you removed the resource group not found part of the message, are these other errors that `CloudError` would expose?

**Review by jborean93 — COMMENTED** [review]
Seems fine to me but just wanted to see why you actual failure message now changes on `CloudError`.

File: lib/ansible/module_utils/azure_rm_common.py (hunk 1 above)
Comment (ZhijunZhao) [reply lib/ansible/module_utils/azure_rm_common.py:419]: I think the origin error message is too subjective. There could be other reasons causing the exception. 

Yes, I think `CloudError` can give us accurate error message. With the new change, the error is as below if using a non-existing resource group. 


```
TASK [create storage account for function apps] *********************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error retrieving resource group zhijzhao4 - Resource group 'zhijzhao4' could not be found."}
```

Comment (jborean93) [reply lib/ansible/module_utils/azure_rm_common.py:419]: Sounds good to me, just wanted to check.

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
rebuild_merge

**Pull request merged**